### PR TITLE
added dependency on plugin utils, to open more projects correctly

### PR DESCRIPTION
**Pull request by Olenek**

### CODE REVIEW

File: settings.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import java.net.URI
+
+rootProject.name = "extract-method-experiments"
```
Comment (onewhl) [inline settings.gradle.kts:3]: should we define group and version here as well? 
```
group = "org.jetbrains.research.extractMethod"
version = "1.0"
```

File: build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,59 @@
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+group = rootProject.group
```
Comment (onewhl) [inline build.gradle.kts:3]: should we define group and version here as well?

```
group = "org.jetbrains.research.extractMethod"
version = "1.0"
```
it's a root project isn't it?

File: build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+version = rootProject.version
+
+fun properties(key: String) = project.findProperty(key).toString()
+
+plugins {
+    java
+    kotlin("jvm") version "1.5.21" apply true
+    id("org.jetbrains.intellij") version "1.1.3" apply true
+}
```
Comment (onewhl) [inline build.gradle.kts:12]: add a new line

File: settings.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    gitRepository(URI.create(utilitiesRepo)) {
+        producesModule("$utilitiesProjectName:plugin-utilities-core")
+    }
+}
+
+pluginManagement {
+    repositories {
+        gradlePluginPortal()
+        jcenter()
+        maven(url = "https://nexus.gluonhq.com/nexus/content/repositories/releases")
```
Comment (onewhl) [inline settings.gradle.kts:21]: do we need it?

File: extract-method-plugin/src/main/kotlin/org/jetbrains/research/extractMethod/BaseRunner.kt [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+import org.jetbrains.research.extractMethod.core.extractors.NegativeRefactoringsExtractionRunner
+import org.jetbrains.research.extractMethod.core.extractors.PositiveRefactoringsExtractionRunner
+import org.jetbrains.research.pluginUtilities.openRepository.getKotlinJavaRepositoryOpener
+import org.jetbrains.research.pluginUtilities.preprocessing.getKotlinJavaPreprocessorManager
+import java.io.FileWriter
+import java.nio.file.Files
+import java.nio.file.Path
+import kotlin.streams.toList
+
+private val preprocessor = getKotlinJavaPreprocessorManager(null)
```
Comment (onewhl) [inline extract-method-plugin/src/main/kotlin/org/jetbrains/research/extractMethod/BaseRunner.kt:13]: why it is defined here, not inside the class?

File: extract-method-plugin/src/main/java/org/jetbrains/research/extractMethod/PluginRunner.java [hunk 6]
```diff
@@ -35,18 +37,31 @@ public void main(@NotNull List<String> args) {
...
         runExtractions(line);
     }
 
     private void runExtractions(CommandLine cmdLine) {
-        List<String> projectPaths = new ArrayList<>();
         StringBuilder outputDirPathBuilder = new StringBuilder();
 
-        configureIO(projectPaths, outputDirPathBuilder, cmdLine);
+        BaseRunner runner = new BaseRunner();
+        try{
```
Comment (onewhl) [inline extract-method-plugin/src/main/java/org/jetbrains/research/extractMethod/PluginRunner.java:51]: add a space after `try`

File: extract-method-plugin/src/main/java/org/jetbrains/research/extractMethod/PluginRunner.java [hunk 7]
```diff
@@ -7,17 +7,19 @@
...
 import java.io.File;
 import java.io.FileWriter;
 import java.io.IOException;
 import java.nio.file.Files;
+import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.ArrayList;
 import java.util.List;
 
+import org.jetbrains.research.extractMethod.BaseRunnerKt;
```
Comment (onewhl) [inline extract-method-plugin/src/main/java/org/jetbrains/research/extractMethod/PluginRunner.java:21]: unused import

File: extract-method-plugin/src/main/java/org/jetbrains/research/extractMethod/PluginRunner.java [hunk 8]
```diff
@@ -7,17 +7,19 @@
 import org.jetbrains.annotations.NonNls;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.research.extractMethod.core.extractors.NegativeRefactoringsExtractionRunner;
```
Comment (onewhl) [inline extract-method-plugin/src/main/java/org/jetbrains/research/extractMethod/PluginRunner.java:9]: unused import

File: extract-method-plugin/src/main/kotlin/org/jetbrains/research/extractMethod/BaseRunner.kt [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+    private val logger = LogManager.getLogger(BaseRunner::class)
+
+    private fun getSubdirectories(path: Path): List<Path> {
+        return Files.walk(path, 1)
+            .filter { Files.isDirectory(it) && !it.equals(path) }
+            .toList()
+    }
+
+    fun runPositives(inputDir : Path, fw : FileWriter) {
+        val runner = PositiveRefactoringsExtractionRunner(fw);
```
Comment (onewhl) [inline extract-method-plugin/src/main/kotlin/org/jetbrains/research/extractMethod/BaseRunner.kt:26]: what about extracting base interface named `RefactoringExtractor` and add two implementations -- one for `PositiveRefactoringsExtractor` and another one for `NegativeRefactoringsExtractor`? 
they both have methods with the same names.

File: extract-method-plugin/src/main/kotlin/org/jetbrains/research/extractMethod/BaseRunner.kt [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+import org.jetbrains.research.pluginUtilities.preprocessing.getKotlinJavaPreprocessorManager
+import java.io.FileWriter
+import java.nio.file.Files
+import java.nio.file.Path
+import kotlin.streams.toList
+
+private val preprocessor = getKotlinJavaPreprocessorManager(null)
+private val repositoryOpener = getKotlinJavaRepositoryOpener()
+
+class BaseRunner {
```
Comment (onewhl) [inline extract-method-plugin/src/main/kotlin/org/jetbrains/research/extractMethod/BaseRunner.kt:16]: maybe we should name it as `RepositoryOpener`?
also, these two methods look similar. we can just leave only one and add a new parameter `extractor: RefactoringExtractor` (see the comment below about common interface for refactoring extractors)

File: extract-method-plugin/src/main/kotlin/org/jetbrains/research/extractMethod/BaseRunner.kt (hunk 9 above)
Comment (Olenek) [reply extract-method-plugin/src/main/kotlin/org/jetbrains/research/extractMethod/BaseRunner.kt:26]: Was about to do it today, actually!

File: extract-method-plugin/src/main/java/org/jetbrains/research/extractMethod/PluginRunner.java (hunk 8 above)
Comment (Olenek) [reply extract-method-plugin/src/main/java/org/jetbrains/research/extractMethod/PluginRunner.java]: removed

File: extract-method-plugin/src/main/java/org/jetbrains/research/extractMethod/PluginRunner.java [hunk 11]
```diff
@@ -66,31 +66,42 @@ private void runExtractions(CommandLine cmdLine) {
         String outputDirPath = outputDirPathBuilder.toString();
 
         if (cmdLine.hasOption("generatePositiveSamples")) {
-            FileWriter positiveFW = null;
+            FileWriter positiveFW;
+            RefactoringsExtractor extractor;
             try {
                 positiveFW = makePositiveHeader(outputDirPath);
```
[the remark this answers is not in the document]
Comment (Olenek) [reply extract-method-plugin/src/main/java/org/jetbrains/research/extractMethod/PluginRunner.java]: addressed it

File: extract-method-plugin/src/main/kotlin/org/jetbrains/research/extractMethod/BaseRunner.kt [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+    private val preprocessor = getKotlinJavaPreprocessorManager(null)
+    private val repositoryOpener = getKotlinJavaRepositoryOpener()
+
+    private fun getSubdirectories(path: Path): List<Path> {
+        return Files.walk(path, 1)
+            .filter { Files.isDirectory(it) && !it.equals(path) }
+            .toList()
+    }
+
+    fun runExtractions(inputDir: Path, extractor: RefactoringsExtractor) {
```
Comment (onewhl) [inline extract-method-plugin/src/main/kotlin/org/jetbrains/research/extractMethod/BaseRunner.kt:22]: please, try to do it in another way ([standardRepositoryOpener](https://github.com/JetBrains-Research/Kotlin-Analysis/blob/5af4060d8be0ef4dc8f79e2beee78a44a4042cad/kotlin-analysis-core/src/main/kotlin/org/jetbrains/research/ml/kotlinAnalysis/util/RepositoryOpenerUtil.kt#L41))

**Pull request merged**